xarray/core/missing.py

```diff
@@ -751,3 +803,76 @@ def _interpnd(var, x, new_x, func, kwargs):
     # move back the interpolation axes to the last position
     rslt = rslt.transpose(range(-rslt.ndim + 1, 1))
     return rslt.reshape(rslt.shape[:-1] + new_x[0].shape)
+
+
+def _dask_aware_interpnd(var, func: Callable[..., Any], kwargs: Any, nx: int, *arrs):
+    """Wrapper for `_interpnd` allowing dask array to be used in `map_blocks`
+
+    The first `nx` arrays in `arrs` are original coordinates, the rest are destination coordinate
+    Currently this need original coordinate to be full arrays (meshgrid)
+
+    TODO: find a way to use 1d coordinates
+    """
+    from .dataarray import DataArray
+
+    _old_x, _new_x = arrs[:nx], arrs[nx:]
+
+    # reshape x (TODO REMOVE)
+    old_x = tuple(
+        [
+            np.moveaxis(tmp, dim, -1)[tuple([0] * (len(tmp.shape) - 1))]
+            for dim, tmp in enumerate(_old_x)
+        ]
+    )
+
+    new_x = tuple([DataArray(_x) for _x in _new_x])
+
+    return _interpnd(var, old_x, new_x, func, kwargs)
+
+
+def _add_interp_ghost(var, x, nconst: int):
```

Comment:
can we use `map_overlap` instead of `_add_interp_ghost + map_blocks`?

Reply:
Not directly because it doesn't pass `*args` through to `map_blocks`

Reply:
I think it does. See https://github.com/dask/dask/blob/137d039250eadf757b8941d6e1c5cfb6c1e61679/dask/array/overlap.py#L714

Is there a specific problem you were running into? Maybe pass them as kwargs?

Reply:
OK, sorry, I answered too fast...
In fact the problem is that I need `var` with overlap, `old_x` with overlap and `new_x` **without** overlap...
And I don't think I can do this without `map_blocks`